Assets/BossRoom/Scripts/Shared/Infrastructure/PubSub/NetworkedMessageChannel.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using Unity.Collections;
+using Unity.Netcode;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.BossRoom.Shared.Infrastructure
+{
+    /// <summary>
+    /// This type of message channel allows the server to publish a message that will be sent to clients as well as
+    /// being published locally. Clients and the server both can subscribe to it, but it needs to be done after
+    /// the NetworkManager is initialized.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class NetworkedMessageChannel<T> : MessageChannel<T> where T : unmanaged
+    {
+        string m_Name;
+
+        int m_BufferSize;
+
+        bool m_HasRegisteredHandler;
+
+        public NetworkedMessageChannel(string name, int bufferSize)
+        {
+            m_Name = name;
```

Comment:
name could be deduced from type T? I would make sure to append "NetworkedMessageChannel" in from of it to make sure there's no future name conflict